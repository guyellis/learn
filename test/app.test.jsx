const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const App = require('../src/app');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('App', () => {
  test('should render an App component', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </MuiThemeProvider>);

    const app = component.toJSON();
    expect(app).toMatchSnapshot();
  });
});
