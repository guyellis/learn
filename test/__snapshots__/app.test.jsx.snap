// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render an App component 1`] = `
<div>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="makeStyles-appLabel-6"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          >
            LEARN
          </h6>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
        >
          <a
            aria-describedby={null}
            aria-label="Fork guyellis/learn on GitHub"
            className=""
            href="https://github.com/guyellis/learn"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Fork me!"
          >
            <img
              alt="github"
              height={30}
              src={Object {}}
              width={30}
            />
          </a>
        </div>
      </div>
    </header>
    <main
      className="makeStyles-content-4"
    >
      <div
        className="makeStyles-toolbar-7"
      />
      <div>
        <div>
          <h2
            className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
          >
            Home
          </h2>
        </div>
      </div>
    </main>
  </div>
</div>
`;
